source/module_hsolver/diag_hs_para.cpp

```diff
@@ -0,0 +1,201 @@
+#include <iostream>
+#include "module_hsolver/diag_hs_para.h"
+#include "module_basis/module_ao/parallel_2d.h"
+#include "module_hsolver/diago_pxxxgvx.h"
+#include "module_base/scalapack_connector.h"
+#include "module_hsolver/genelpa/elpa_solver.h"
+
+namespace hsolver
+{
+
+#ifdef __ELPA 
+    void elpa_diag(MPI_Comm comm,
+                   const int nband, 
+                   std::complex<double>* h_local, 
+                   std::complex<double>* s_local, 
+                   double* ekb, 
+                   std::complex<double>* wfc_2d,
+                   Parallel_2D& para2d_local)
+    {
+        int DecomposedState = 0;
+        ELPA_Solver es(false,
+                   comm,
+                   nband,
+                   para2d_local.get_row_size(),
+                   para2d_local.get_col_size(),
+                   para2d_local.desc);
+        es.generalized_eigenvector(h_local, s_local, DecomposedState, ekb, wfc_2d);
+        es.exit();
+    }
+
+    void elpa_diag(MPI_Comm comm,
+                   const int nband, 
+                   double* h_local, 
+                   double* s_local, 
+                   double* ekb, 
+                   double* wfc_2d,
+                   Parallel_2D& para2d_local)
+    {
+        int DecomposedState = 0;
+        ELPA_Solver es(true,
+                   comm,
+                   nband,
+                   para2d_local.get_row_size(),
+                   para2d_local.get_col_size(),
+                   para2d_local.desc);
+        es.generalized_eigenvector(h_local, s_local, DecomposedState, ekb, wfc_2d);
+        es.exit();
+    }
+
+    void elpa_diag(MPI_Comm comm,
+                   const int nband, 
+                   std::complex<float>* h_local, 
+                   std::complex<float>* s_local, 
+                   float* ekb, 
+                   std::complex<float>* wfc_2d,
+                   Parallel_2D& para2d_local)
+    {
+        std::cout << "Error: ELPA do not support single precision. " << std::endl;
+        exit(1);
+    }
+
+    void elpa_diag(MPI_Comm comm,
+                   const int nband, 
+                   float* h_local, 
+                   float* s_local, 
+                   float* ekb, 
+                   float* wfc_2d,
+                   Parallel_2D& para2d_local)
+    {
+        std::cout << "Error: ELPA do not support single precision. " << std::endl;
+        exit(1);
+    }
+
+#endif
+
+
+#ifdef __MPI
+
+template <typename T>
+void Diago_HS_para(
+                T* h, 
```

Comment:
I would suggset using clang-format before you pr a new code, here T is not aligned with the following variables